### ISSUE TITLE
[BUG] Dictionary Type not working with OpenAI Extractor

### ISSUE THREAD

**Issue by aowen14** [Mirascope/mirascope#168]
Not sure if a bug or a feature request, as it might be related to OpenAI. Extracting to a Pydantic model that contains a dictionary throws validation errors when Pydantic tries to convert the returned string to a model.

**To Reproduce**
Copy and Paste these code blocks into a notebook.
Fill in your API key on this notebook, run the cells.

```
from typing import Dict, List, Type
from pydantic import BaseModel
from mirascope.openai import OpenAICall, OpenAICallParams, OpenAIExtractor
import json
import os

os.environ["OPENAI_API_KEY"] = "<Api-Key>"

class FoodExamples(BaseModel):
    foods_and_examples : Dict[str, List[str]]

class FoodExamplesExtractor(OpenAIExtractor[FoodExamples]):
    extract_schema: Type[FoodExamples] = FoodExamples
    prompt_template = """
    Extract the food types and their specific examples from the following json string:
    {json_string}
    """
    json_string : str 
    call_params = OpenAICallParams(model = "gpt-4")

simple_dict = {
    "cheese" : [
        "parmesean",
        "pecorino",
        "mozzarella"
    ],
    "bread" : [
        "sourdough",
        "whole wheat",
        "rye"
    ]
}

food_examples = FoodExamplesExtractor(json_string=json.dumps(simple_dict)).extract()
assert isinstance(food_examples, FoodExamples)
print(food_examples)
```

**Comment by willbakst** [comment]
```python
import json

from openai import OpenAI


tool_schema = {
    "type": "function",
    "function": {
        "name": "FoodExamples",
        "description": "Correctly formatted and typed parameters extracted from the completion. Must include required parameters and may exclude optional parameters unless present in the text.",
        "parameters": {
            "properties": {
                "foods_and_examples": {
                    "additionalProperties": {
                        "items": {"type": "string"},
                        "type": "array",
                    },
                    "title": "Foods And Examples",
                    "type": "object",
                }
            },
            "required": ["foods_and_examples"],
            "type": "object",
        },
    },
}
client = OpenAI()
simple_dict = {
    "cheese": ["parmesean", "pecorino", "mozzarella"],
    "bread": ["sourdough", "whole wheat", "rye"],
}
response = client.chat.completions.create(
    model="gpt-3.5-turbo-0125",
    messages=[
        {
            "role": "system",
            "content": "If a function call accepts a dictionary argument, make sure to populate that dictionary with the expected (key, value) pairs.",
        },
        {
            "role": "user",
            "content": f"Extract the food types and their specific examples from the following json string:\n{json.dumps(simple_dict)}",
        },
    ],
    tools=[tool_schema],
    tool_choice="auto",  # auto is default, but we'll be explicit
)
print(response.choices[0].message.tool_calls)
# [ChatCompletionMessageToolCall(id='call_uJZ2Lza8H5pxjPraJcFYUgYj', function=Function(arguments='{}', name='FoodExamples'), type='function')]
```

It appears that for some reason dictionary properties fail at the OpenAI level (this returns a tool_call with empty arguments...). I've tried prompt engineering and haven't been able to get it to produce one.

I know that Anthropic Claude is able to output dictionaries, so if necessary to extract an arbitrary dict, try Haiku or Sonnet.

**Comment by aowen14** [comment]
Thanks! Will give that a try

**Comment by willbakst** [comment]
#172 enables the use of JSON mode, which enables OpenAI to call tools with arbitrary dicts. Just set the response format:

```python
call_params = OpenAICallParams(model = "gpt-4", response_format={"type": "json_object"})
```

This will be released shortly in v0.8.3
